### PR TITLE
feat: update neptune

### LINKED ISSUE #930: Upgrade Neptune
### Feature
Our neptune version is very old, we should upgrade it.  
We have two options here, we could use a newer version of neptune or we could use the [sacred plugin](https://docs.neptune.ai/integrations/sacred/#__tabbed_1_2)

### PR DESCRIPTION
**Pull request by sash-a**
* Unpinned neptune version
* Removed old `NeptuneLogger` class
* Use sacred integration for logging to neptune - so neptune has been remove from `Logger` instead it is added to sacred when sacred is setup
* Refactored the common logging stuff from all the files into a method: `get_sacred_exp`
* Removed the `stop_logging` method because sacred should stop the neptune logger now
* Converted the hydra config to a dict right away, this lets us remove the `copy_config` method, plays nicer with typing and neptune doesn't complain

Note to run this locally you need to uninstall neptune-client before you update neptune!

```
pip uninstall neptune-client
pip install neptune[sacred]
```

Closes #930

### CODE REVIEW

File: mava/utils/logger_tools.py [hunk 1]
```diff
@@ -249,17 +96,45 @@ def get_logger() -> logging.Logger:
...
-        return {k: config_copy(v) for k, v in config.items()}
-    elif isinstance(config, list):
-        return [config_copy(v) for v in config]
-    else:
-        return deepcopy(config)
+def get_neptune_logger(cfg: Dict) -> neptune.Run:
+    """Set up neptune logging."""
+    name = cfg["name"]
+    tags = cfg["neptune_tag"]
+    project = cfg["neptune_project"]
```
Comment (OmaymaMahjoub) [inline mava/utils/logger_tools.py:103]: If you can add this to config `neptune_project: "InstaDeep/Mava"` else try to set it to the default value.

File: mava/logger.py [hunk 2]
```diff
@@ -27,7 +27,7 @@
 from mava.utils.logger_tools import Logger, get_experiment_path, should_log
 
 
-def get_logger_tools(logger: Logger, config: Dict) -> Tuple[Callable, Callable]:  # noqa: CCR001
+def get_logger_tools(logger: Logger, config: Dict) -> Callable:  # noqa: CCR001
```
Comment (OmaymaMahjoub) [inline mava/logger.py:30]: You can remove `# noqa: CCR001` as this mypy error won't exist after these changes

**Review by OmaymaMahjoub — CHANGES_REQUESTED** [review]
Thanks @sash-a  looking good to me, just minor fixes to be done :pray:

File: mava/logger.py (hunk 2 above)
Comment (sash-a) [reply mava/logger.py:30]: Seems like it's still failing for me if I remove this :cry:

Comment (OmaymaMahjoub) [reply mava/logger.py:30]: yeah I recheck it and it doesn't work unfortunately :(

File: mava/systems/ff_mappo_rware.py [hunk 3]
```diff
@@ -591,27 +590,14 @@ def run_experiment(_run: run.Run, _config: Dict, _log: SacredLogger) -> None:
...
-    # Set the base path for the experiment.
-    cfg["system_name"] = "ff_mappo"
-    exp_path = get_experiment_path(cfg, "sacred")
-    file_obs_path = os.path.join(cfg["base_exp_path"], exp_path)
-    ex.observers = [observers.FileStorageObserver.create(file_obs_path)]
-
-    # Add configuration to the experiment.
-    ex.add_config(OmegaConf.to_container(cfg, resolve=True))
+    # Convert config to python dict.
+    cfg: Dict = OmegaConf.to_container(cfg, resolve=True)  # type: ignore
```
Comment (OmaymaMahjoub) [inline mava/systems/ff_mappo_rware.py:598]: No need for `type: ignore`

**Review by OmaymaMahjoub — APPROVED** [review]
LGTM! Minor edit that we can skip in case it is unnecessary

**Review by RuanJohn — CHANGES_REQUESTED** [review]
Really nice update @sash-a :raised_hands: 
Just a minor comment here and there.

File: mava/systems/rec_mappo_rware.py [hunk 4]
```diff
@@ -789,27 +788,14 @@ def run_experiment(_run: run.Run, _config: Dict, _log: SacredLogger) -> None:
...
-    exp_path = get_experiment_path(cfg, "sacred")
-    file_obs_path = os.path.join(cfg["base_exp_path"], exp_path)
-    ex.observers = [observers.FileStorageObserver.create(file_obs_path)]
-
-    # Add configuration to the experiment.
-    ex.add_config(OmegaConf.to_container(cfg, resolve=True))
+    # Convert config to python dict.
+    cfg: Dict = OmegaConf.to_container(cfg, resolve=True)  # type: ignore
+
+    ex = get_sacred_exp(cfg, "rec_mappo_rware")
```
[the remark this answers is not in the document]
Comment (OmaymaMahjoub) [reply mava/systems/rec_mappo_rware.py:798]: Same really like it :fire:

**Review by OmaymaMahjoub — DISMISSED** [review]
LGTM!

File: mava/utils/logger_tools.py [hunk 5]
```diff
@@ -249,17 +96,51 @@ def get_logger() -> logging.Logger:
...
+    return run
+
+
+def get_sacred_exp(cfg: Dict, system_name: str) -> Experiment:
+    """Get sacred experiment and set up sacred logging.
+
+    This sets up terminal logging, adds the file observer (to log configs as json files)
+    and neptune logging (to save logs online) if required.
+
+    Stores files at: base_exp_path/system_name/env_name/task_name/envs_num_envs/seed_seed.
```
Comment (RuanJohn) [inline mava/utils/logger_tools.py:119]:
```suggestion
    Stores files at: base_exp_path/system_name/env_name/task_name/num_envs/seed.
```

File: mava/utils/logger_tools.py [hunk 6]
```diff
@@ -249,17 +96,51 @@ def get_logger() -> logging.Logger:
...
+    del cfg["neptune_tag"]  # neptune doesn't want lists in run params
+    run["params"] = cfg
+
+    return run
+
+
+def get_sacred_exp(cfg: Dict, system_name: str) -> Experiment:
+    """Get sacred experiment and set up sacred logging.
+
+    This sets up terminal logging, adds the file observer (to log configs as json files)
```
Comment (RuanJohn) [inline mava/utils/logger_tools.py:116]:
```suggestion
    This sets up terminal logging, adds the file observer (to log configs and results as json files)
```

**Pull request merged**